### PR TITLE
Evo 1120 farm filters

### LINKED ISSUE #1120: Group explorer: Farm group type filtering
Add filters by various fields in the farm schema to the group explorer when farm type is selected

### PR DESCRIPTION
**Pull request by thomasgwatson**
Closes #1120 

image

image

### CODE REVIEW

**Comment by tibetsprague** [comment]
can you rebase farm on dev so I dont see all the extra changes?

**Comment by brodeur** [comment]
OK Pushed my stuff... the only thing remaining that I can see that's a bug is that if I open the filters, and the switch to the "all" groups view, the filter list remains open.

File: src/css/global/icons.scss [hunk 1]
```diff
@@ -283,3 +283,24 @@
...
+.icon-Filter:before {
+  content: "\e95a";
+}
+.icon-Mentorship:before {
+  content: "\e95b";
+}
+.icon-Message:before {
+  content: "\e95c";
+}
+.icon-Volunteer:before {
```
Comment (tibetsprague) [inline src/css/global/icons.scss:301]: what the heck is up with the new files that end with Zone.identifier? like public/assets/fonts/icons/hylo-evo-icons.eot:Zone.Identifier

File: src/routes/GroupExplorer/GroupSearch/GroupSearch.js [hunk 2]
```diff
@@ -27,9 +30,15 @@ export default function GroupSearch () {
   const [sortBy, setSortBy] = useState(SORT_NAME)
   const [search, setSearch] = useState('')
   const [offset, setOffset] = useState(0)
+  const [filterToggle, setFilterToggle] = useState(false)
+  const [groupType, setGroupType] = useState(null)
+  const [farmType, setFarmType] = useState('')
```
Comment (tibetsprague) [inline src/routes/GroupExplorer/GroupSearch/GroupSearch.js:35]: why do we store farmType, certOrManagementPlan and producrCategories stored twice in the state, both individually and in farmQuery? seems like we could we just have it in farmQuery?

File: src/routes/GroupExplorer/GroupSearch/GroupSearch.js [hunk 3]
```diff
@@ -40,61 +49,77 @@ export default function GroupSearch () {
     search: debouncedSearchTerm,
     offset,
     nearCoord,
-    visibility: [3]
+    visibility: [3],
+    groupType,
```
Comment (tibetsprague) [inline src/routes/GroupExplorer/GroupSearch/GroupSearch.js:53]: still like to alphabetize lists

File: src/store/actions/fetchGroups.js [hunk 4]
```diff
@@ -16,11 +16,13 @@ export function fetchGroups ({ offset, order, search, slug, sortBy, nearCoord, p
...
       variables: {
         first: pageSize,
+        farmQuery,
         nearCoord,
         offset: offset,
         order,
         search,
-        sortBy
+        sortBy,
+        groupType
```
Comment (tibetsprague) [inline src/store/actions/fetchGroups.js:25]: alpha

File: src/store/actions/fetchGroups.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ import { makeGetQueryResults, makeQueryResultsModelSelector } from 'store/reduce
 export const MODULE_NAME = 'FeedList'
 export const STORE_FETCH_POSTS_PARAM = `${MODULE_NAME}/STORE_FETCH_POSTS_PARAM`
 
-export function fetchGroups ({ offset, order, search, slug, sortBy, nearCoord, pageSize = 20 }) {
+export function fetchGroups ({ nearCoord, offset, order, pageSize = 20, search, slug, sortBy, groupType, farmQuery }) {
```
Comment (tibetsprague) [inline src/store/actions/fetchGroups.js:8]: alpha

File: src/store/actions/fetchGroups.js [hunk 6]
```diff
@@ -37,20 +39,24 @@ const groupQuery = `
...
   $boundingBox: [PointInput],
   $first: Int,
+  $farmQuery: JSON,
   $nearCoord: PointInput,
   $offset: Int,
   $order: String,
   $search: String,
-  $sortBy: String
+  $sortBy: String,
+  $groupType: String
```
Comment (tibetsprague) [inline src/store/actions/fetchGroups.js:48]: alpha

File: src/util/constants.js [hunk 7]
```diff
@@ -1 +1,2713 @@
...
+    label: 'wheat',
+    value: 'wheat',
+    id: '6087e76c53709500013da9b9'
+  },
+  {
+    label: 'durum wheat',
+    value: 'wheat_durum',
+    id: '6087e76c53709500013da9ba'
+  },
+  {
```
Comment (tibetsprague) [inline src/util/constants.js:2693]: should this be in its own farm_schema file or something?

File: src/util/index.js [hunk 8]
```diff
@@ -37,3 +37,7 @@ export const validateEmail = email => {
   const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
   return re.test(email.toLowerCase())
 }
+
+export function isDateInTheFuture (date) {
```
Comment (tibetsprague) [inline src/util/index.js:41]: i think we use to have this and then i think @lorenjohnson moved it to hylo-shared?

File: src/css/global/icons.scss (hunk 1 above)
Comment (thomasgwatson) [reply src/css/global/icons.scss:301]: yeah I don't know what they are for but they came along with the other files I was expecting... ahh it looks like its a browser metadata file added when I downloaded them. I'll delete

File: src/util/index.js (hunk 8 above)
Comment (thomasgwatson) [reply src/util/index.js:41]: righto, I think it got dragged back in during the merge deconfliction (which was very hairy for this one...)

**Comment by thomasgwatson** [comment]
Yep sorry, should be good now

**Pull request merged**